### PR TITLE
Fix #414: Add StringToFractionParserTest

### LINKED ISSUE #414: Add tests for whitespace & fractions patterns
See #413 for details.

### PR DESCRIPTION
**Pull request by viktoriias**
## Explanation

Fixes #414: Add unit tests for StringToFractionParser.

## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

**Comment by viktoriias** [comment]
@BenHenning Please review.
I closed https://github.com/oppia/oppia-android/pull/3112 because of sync issues, but your comments are addressed.
Checks here are failing because of https://github.com/oppia/oppia-android/issues/3157

**Comment by BenHenning** [comment]
Sorry, been really busy with non-work items this week. Will need to take a look later in the week.

File: app/src/test/java/org/oppia/android/app/parser/StringToFractionParserTest.kt [hunk 1]
```diff
@@ -0,0 +1,462 @@
...
+  @Test
+  fun testParser_realTimeError_severalDashes_invalidFormat_hasRelevantErrorMessage() {
+    val errorMessage = stringToFractionParser.getRealTimeAnswerError("-1/-3")
+      .getErrorMessageFromStringRes(context)
+    assertThat(errorMessage)
+      .isEqualTo("Please enter a valid fraction (e.g., 5/3 or 1 2/3)")
+  }
+
+  @Test
+  fun testParser_parseFraction_divisionByZero_parsesFraction() {
```
Comment (BenHenning) [inline app/src/test/java/org/oppia/android/app/parser/StringToFractionParserTest.kt]: Here & below: I suggest also keeping clear condition/outcomes. E.g.:

- ``testParseFraction_divisionByZero_returnsFraction``
- ``testParseFraction_multipleFractions_failsWithError``
- ``testParseFraction_regularFraction_returnsFraction``

etc.

**Review by BenHenning — CHANGES_REQUESTED** [review]
Thanks @viktoriias! I think this looks really good. Just had one nit on test naming--PTAL.

File: app/src/test/java/org/oppia/android/app/parser/StringToFractionParserTest.kt (hunk 1 above)
Comment (viktoriias) [reply app/src/test/java/org/oppia/android/app/parser/StringToFractionParserTest.kt]: Done. Renamed here, below and some above.

**Review by rt4914 — APPROVED** [review]
LGTM, thanks.

**Comment by viktoriias** [comment]
@BenHenning PTAL

**Review by BenHenning — APPROVED** [review]
Apologies @viktoriias I've been traveling this week. I think the updated test names look excellent! The PR LGTM, thanks!
